js/data.js

```diff
@@ -0,0 +1,321 @@
+const rickAndMortyApi = "https://rickandmortyapi.com/api/character";
+const URL = "https://reqres.in/api/users?page=1";
+
+const isVowelFilterActive = () =>
+  document.getElementById("btn-vowel").innerHTML !== "Names starting with a vowel";
+
+const isAlphabetFilterActive = () =>
+  document.getElementById("btn-alphabet").innerHTML !== "Put in alphabetical order";
+
+const isLengthFilterActive = () =>
+  document.getElementById("btn-length").innerHTML !== "Names of five letters or less";
+
+const isReplaceFilterActive = () =>
+  document.getElementById("btn-replace").innerHTML !== "Wubba lubba dub dub";
+
+let select = document.createElement("select");
+select.setAttribute("id", "entries");
+
+function renderSelectList(list, attributeName) {
+  // console.log("list", list);
+  if (document.getElementById("entries")) {
+    document.getElementById("entries").remove();
+  }
+  
+  document.getElementById("select-top-entry").appendChild(select);
+
+  list.map((item) => {
+    let option = document.createElement("option");
+    select.appendChild(option);
+
+    option.innerHTML = item[attributeName];
+
+    option.className = "";
+  });
+
+  select.addEventListener("change", () => selectWho(list, attributeName, select));
+}
+
+// select.addEventListener("change", selectWho); // other way= of writing it
+/** @param  {Array.<object>} list */
+function selectWho(list, attributeName, select) {
+  console.log("select.value", select.value);
+  console.log("select", select);
+  const topUser = list.find((listItem) => {
+    if (listItem[attributeName] === select.value) {
+      return true;
+    } return false;
+  });
+  const filteredList = list.filter((item) =>  {
+    if (item[attributeName] === select.value) {
+      return false;
+    } return true;
+  });
+  console.log("topUser", topUser);
+  console.log("filteredList", filteredList);
+  renderUserList({users:[topUser].concat(filteredList)})
+  // renderUserList({users:[topUser, ...filteredList]})
+}
+
+// function renderSelectedUser({users, selectUser}) {
+//   if (selectUser) {
+//     filteredUsers = filteredUsers.filter((user)) =>
+
+//   }
+// }
+
+function renderUserList({users, filterVowel, filterAlphabet, filterLength}) {
```

Comment:
Fusionner cette fonction avec la fonction renderCharacterList. Pour t'aider tu peux t'inspirer de la fonction renderSelectList qui fonctionne aussi bine pour les users que les characters

Reply:
Soooort of done